metrics-operator/controllers/common/providers/dynatrace/common.go

```diff
@@ -24,20 +25,37 @@ type Error struct {
 	Message string `json:"message"`
 }
 
-func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) (string, error) {
+type SecretValues struct {
+	Token   string `json:"token"`
+	AuthUrl string `json:"authurl"`
+}
+
+func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) ([]byte, error) {
 	if !provider.HasSecretDefined() || !provider.HasSecretKeyDefined() {
-		return "", ErrSecretKeyRefNotDefined
+		return []byte{}, ErrSecretKeyRefNotDefined
 	}
 	dtCredsSecret := &corev1.Secret{}
 	if err := k8sClient.Get(ctx, types.NamespacedName{Name: provider.Spec.SecretKeyRef.Name, Namespace: provider.Namespace}, dtCredsSecret); err != nil {
-		return "", err
+		return []byte{}, err
 	}
 
 	token := string(dtCredsSecret.Data[provider.Spec.SecretKeyRef.Key])
+	secretValues := SecretValues{
```

Comment:
please double check this - ideally with a new unit test. But I believe this will set the `Token` to the value stored in the secret, which will be a JSON object containing the token and the auth URL.
Also, I think for backwards compatibility we should check if the secret value has been stored in the new format (i.e. the JSON object), or if it is still just the token